### ISSUE TITLE
mat-dialog-close wrongly adds an aria-label when the button text content is bound rather than static

### ISSUE THREAD

**Issue by stevenyxu** [angular/material2#15542]
#### What is the expected behavior?
When I have `<button mat-dialog-close>Submit</button>` in a template, I expect for the resulting HTML to be basically the same, and mat-dialog-close should not add an aria label to the button.

#### What is the current behavior?
`<button mat-dialog-close aria-label="Close dialog">Submit</button>`

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-j9paiz?file=app/dialog-overview-example-dialog.html

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Head.

#### Is there anything else we should know?
This area was partly addressed in https://github.com/angular/material2/pull/11093 but has this remaining bug. A workaround is available (just set aria-label explicitly in addition to the inner content), but this is not ideal.

https://github.com/stevenyxu/material2/commit/66453094e69e806f6fb84b2c856baa1ad1c71118 demonstrates the bug in the form of a failing unit test. I tried to do a fix but couldn't quickly come up with a performant solution without more work.

**Comment by jelbourn** [comment]
@crisbeto we should probably just use a mutation observer here
